### DISCUSSION TITLE
Schema for Referral Application

### DISCUSSION THREAD

**Discussion by heiheihang** [LalaCuteGroup/lala-referral#10] [Ideas]
- `Application ID` (string) : Partition Key
- `Attempt (Number)` : Sort Key
- `Applicant TG ID` (string)
- `Referrer TG ID` (string)
- `Resume` (String - to S3 storage)
- `Status` (Number - enum)

Each application has a unique ID (generated by applicantID-ReferrerID as partition Key and attempt number as sort key).
 When an applicant upload a resume, TG Bot will upload it to S3 and retrieve an internal S3 path, which will be stored in this table.
Status is an enum:
- `1` : Waiting for Applicant Information (Resume, Leetcode Verification)
- `2`: Waiting for Referrer Confirmation
- `3`: Need Additional Information
- `4`: Terminated
- `5`: Complete
![Screenshot 2022-07-11 at 10 06 04 PM](https://user-images.githubusercontent.com/24492138/178392963-538899c6-ed4f-4ddb-aa74-078706eb23d7.png)

**Comment by heiheihang** [comment]
@jonowo pointed out there is a duplication of resume in this table and the applicant table. The applicant table is the default resume, and she may want to use another resume for a specific application. I will be editing the one in the applicant table to be "default-resume"

**Comment by jonowo** [comment]
Will we perform leetcode verification on user registration or on first referral application? If we adopt the former, then we don't need to include leetcode verification in the referral application process. We only need to perform leetcode problem count / contest rating check for referral applications.

ඞඞඞ

**Reply by heiheihang** [reply]
I think we should allow verification to be done in both occasions
